app/controllers/mixins/ems_common.rb

```diff
@@ -193,15 +193,18 @@ def ad_hoc_metrics_pressed
     def block_storage_manager_id(provider_id)
       manager = find_record_with_rbac(ExtManagementSystem, provider_id)
       return nil unless manager
-      return manager.id unless manager.respond_to?(:storage_managers)
+      return manager.id unless manager.respond_to?(:storage_managers) || manager.respond_to?(:storage_manager)
```

Comment:
I think `supports?` is preferred over `respond_to?` /cc @agrare

Reply:
Yes, this is also a lot of logic to find the block storage manager.  The fact that some providers have multiple storage managers and some have one isn't something the UI should have to deal with.

Thinking if a provider has storage managers (need a supports feature for this) then it should have a method `#block_storage_manager` to return to you the appropriate block storage manager or `nil` if there isn't one